File: widgetssdk/src/main/java/com/glia/widgets/core/permissions/domain/HasScreenSharingNotificationChannelEnabledUseCase.java
package com.glia.widgets.core.permissions.domain;

import com.glia.widgets.core.permissions.PermissionManager;

public class HasScreenSharingNotificationChannelEnabledUseCase {
    private final PermissionManager permissionManager;

    public HasScreenSharingNotificationChannelEnabledUseCase(PermissionManager permissionManager) {
        this.permissionManager = permissionManager;
    }

    public boolean execute() {
        return permissionManager.hasScreenSharingNotificationChannelEnabled();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/permissions/domain/HasScreenSharingNotificationChannelEnabledUseCase.kt
package com.glia.widgets.core.permissions.domain

import android.content.Context
import com.glia.widgets.core.notification.NotificationFactory
import com.glia.widgets.core.notification.areNotificationsEnabledForChannel

internal class HasScreenSharingNotificationChannelEnabledUseCase(private val context: Context) {
    operator fun invoke(): Boolean = context.areNotificationsEnabledForChannel(NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID)
}
